src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        if (action.extraData) {
+          Object.assign(stateCopy, action.extraData);
+        }
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
```

Comment:
you can do like so, it could be easier. But anyway, good job.
```
 case 'removeProperties':
        action.keysToRemove.forEach(key => delete stateCopy[key]);
        break;
```